Cannot docker run

#### Environment:
OS: Win
Plugins Version: Azure java toolkit for IntelliJ: 3.37.0
IDE Version: IU2019.3-jre11
#### Repro steps:
1. Right click your project select Azure > Docker run
#### Result:
![image](https://user-images.githubusercontent.com/46642771/83216311-8ca75200-a19b-11ea-90f4-d73bb77757c9.png)
Error logs:
```
java.lang.IllegalStateException: Fatal Exception thrown on Scheduler.Worker thread.
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:59)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.NoSuchMethodError: 'com.google.common.collect.ImmutableList com.spotify.docker.client.messages.Container.ports()'
	at com.microsoft.intellij.runner.container.dockerhost.DockerHostRunState.executeSteps(DockerHostRunState.java:151)
	at com.microsoft.intellij.runner.container.dockerhost.DockerHostRunState.executeSteps(DockerHostRunState.java:55)
	at com.microsoft.intellij.runner.AzureRunProfileState.lambda$execute$0(AzureRunProfileState.java:70)
	at rx.internal.operators.OnSubscribeFromCallable.call(OnSubscribeFromCallable.java:48)
	at rx.internal.operators.OnSubscribeFromCallable.call(OnSubscribeFromCallable.java:33)
	at rx.Observable.unsafeSubscribe(Observable.java:10327)
	at rx.internal.operators.OperatorSubscribeOn$SubscribeOnSubscriber.call(OperatorSubscribeOn.java:100)
	at rx.internal.schedulers.CachedThreadScheduler$EventLoopWorker$1.call(CachedThreadScheduler.java:230)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	... 6 more
```

#### Expected:
![image](https://user-images.githubusercontent.com/46642771/83216127-1acf0880-a19b-11ea-9f4b-365fe91dce5d.png)

--- a/PluginsAndFeatures/azure-toolkit-for-intellij/build.gradle
+++ b/PluginsAndFeatures/azure-toolkit-for-intellij/build.gradle
@@ -113,7 +113,6 @@ dependencies {
     }
     compile 'com.microsoft.azure:azure-tools-common:0.2.0'
     compile group: 'com.microsoft.azure.appplatform.v2019_05_01_preview', name: 'azure-mgmt-appplatform', version: '1.0.0-beta-2'
-    compile 'com.spotify:docker-client:8.16.0'
 
     compile group: 'org.dom4j', name: 'dom4j', version: '2.1.3'
     compile group: 'jaxen', name: 'jaxen', version: '1.2.0'

--- a/PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/dockerhost/DockerHostRunState.java
+++ b/PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/container/dockerhost/DockerHostRunState.java
@@ -25,7 +25,6 @@ package com.microsoft.intellij.runner.container.dockerhost;
 import static com.microsoft.azuretools.telemetry.TelemetryConstants.DEPLOY_WEBAPP_DOCKERLOCAL;
 import static com.microsoft.azuretools.telemetry.TelemetryConstants.WEBAPP;
 
-import com.google.common.collect.ImmutableList;
 import com.intellij.execution.process.ProcessEvent;
 import com.intellij.execution.process.ProcessListener;
 import com.intellij.execution.process.ProcessOutputTypes;
@@ -43,6 +42,7 @@ import com.microsoft.intellij.util.MavenRunTaskUtil;
 import com.spotify.docker.client.DockerClient;
 import com.spotify.docker.client.messages.Container;
 
+import com.spotify.docker.client.shaded.com.google.common.collect.ImmutableList;
 import org.jetbrains.annotations.NotNull;
 
 import java.io.FileNotFoundException;

--- a/PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/WebAppSlimSettingPanel.java
+++ b/PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/WebAppSlimSettingPanel.java
@@ -271,9 +271,11 @@ public class WebAppSlimSettingPanel extends AzureSettingPanel<WebAppConfiguratio
             lblCreateWebApp.setVisible(false);
             cbxWebApp.setVisible(true);
             cbxWebApp.addItem(CREATE_NEW_WEBAPP);
+            // Need calculated target id first or fill combo box will trigger event to change selectedWebApp
             final String configurationWebAppId = webAppConfiguration.getWebAppId();
             final String targetId = (StringUtils.isEmpty(configurationWebAppId) && selectedWebApp != null) ?
                                     selectedWebApp.getResource().id() : configurationWebAppId;
+            sortedWebAppLists.forEach(webAppResourceEx -> cbxWebApp.addItem(webAppResourceEx));
             final ResourceEx<WebApp> selectWebApp = sortedWebAppLists
                     .stream()
                     .filter(webAppResourceEx -> StringUtils.equals(webAppResourceEx.getResource().id(), targetId))

--- a/PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/WebAppSlimSettingPanel.java
+++ b/PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/WebAppSlimSettingPanel.java
@@ -271,17 +271,18 @@ public class WebAppSlimSettingPanel extends AzureSettingPanel<WebAppConfiguratio
             lblCreateWebApp.setVisible(false);
             cbxWebApp.setVisible(true);
             cbxWebApp.addItem(CREATE_NEW_WEBAPP);
-            sortedWebAppLists.forEach(webAppResourceEx -> cbxWebApp.addItem(webAppResourceEx));
-            // Find webapp which id equals to selected web app(when cancel creation) or configuration
-            // Use the first one as fallback
-            final String targetId = selectedWebApp == null ? webAppConfiguration.getWebAppId() :
-                                    selectedWebApp.getResource().id();
-            final ResourceEx<WebApp> selectWebApp = sortedWebAppLists
-                    .stream()
-                    .filter(webAppResourceEx -> StringUtils.equals(webAppResourceEx.getResource().id(), targetId))
-                    .findFirst()
-                    .orElse(sortedWebAppLists.get(0));
-            cbxWebApp.setSelectedItem(selectWebApp);
+            String targetId = webAppConfiguration.getWebAppId();
+            if (StringUtils.isEmpty(targetId) && selectedWebApp != null) {
+                // Use when user cancel create app service
+                targetId = selectedWebApp.getResource().id();
+            }
+            for (int i = 0; i < sortedWebAppLists.size(); i++) {
+                final ResourceEx<WebApp> webAppResourceEx = sortedWebAppLists.get(i);
+                cbxWebApp.addItem(webAppResourceEx);
+                if (i == 0 || StringUtils.equals(webAppResourceEx.getResource().id(), targetId)) {
+                    cbxWebApp.setSelectedItem(webAppResourceEx);
+                }
+            }
         }
         selectWebApp();
         cbxWebApp.setEnabled(true);

--- a/PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/WebAppSlimSettingPanel.java
+++ b/PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/WebAppSlimSettingPanel.java
@@ -271,18 +271,15 @@ public class WebAppSlimSettingPanel extends AzureSettingPanel<WebAppConfiguratio
             lblCreateWebApp.setVisible(false);
             cbxWebApp.setVisible(true);
             cbxWebApp.addItem(CREATE_NEW_WEBAPP);
-            String targetId = webAppConfiguration.getWebAppId();
-            if (StringUtils.isEmpty(targetId) && selectedWebApp != null) {
-                // Use when user cancel create app service
-                targetId = selectedWebApp.getResource().id();
-            }
-            for (int i = 0; i < sortedWebAppLists.size(); i++) {
-                final ResourceEx<WebApp> webAppResourceEx = sortedWebAppLists.get(i);
-                cbxWebApp.addItem(webAppResourceEx);
-                if (i == 0 || StringUtils.equals(webAppResourceEx.getResource().id(), targetId)) {
-                    cbxWebApp.setSelectedItem(webAppResourceEx);
-                }
-            }
+            final String configurationWebAppId = webAppConfiguration.getWebAppId();
+            final String targetId = (StringUtils.isEmpty(configurationWebAppId) && selectedWebApp != null) ?
+                                    selectedWebApp.getResource().id() : configurationWebAppId;
+            final ResourceEx<WebApp> selectWebApp = sortedWebAppLists
+                    .stream()
+                    .filter(webAppResourceEx -> StringUtils.equals(webAppResourceEx.getResource().id(), targetId))
+                    .findFirst()
+                    .orElse(sortedWebAppLists.get(0));
+            cbxWebApp.setSelectedItem(selectWebApp);
         }
         selectWebApp();
         cbxWebApp.setEnabled(true);

